syntax = "proto3";

package common.v4;

option go_package = ".;pb";

message Timestamp {
  int64 unix_ms = 1;
}

message RequestMeta {
  string request_id = 1;
  Timestamp time = 2;
}

message Response {
    RequestMeta meta = 1;
    string message = 2;
}

message PlanStep {
  int32 step_id = 1;
  string action = 2;
  string service = 3;
  map<string, string> input = 4;
  string expected_output = 5;
}

message Plan {
  string goal = 1;
  repeated PlanStep steps = 2;
  repeated string constraints = 3;
  map<string, string> metadata = 4;
}
